src/types.ts

```diff
@@ -50,12 +51,15 @@ export type WidgetTransform<S> = {
 };
 
 export type FormatterSettings = {
```

Comment:
IMO we should set `FormatterSettings` as:
```
type FormatterSettings = NumericFormatter | DateFormatter | StringFormatter;
```

Right now everything is mixed and it's hard to know what to expect in particular use case.